src/stream/src/executor/backfill/cdc/upstream_table/snapshot.rs

```diff
@@ -104,9 +107,28 @@ impl UpstreamTableRead for UpstreamTableReader<ExternalStorageTable> {
 
         let mut builder = DataChunkBuilder::new(self.inner.schema().data_types(), args.chunk_size);
         let chunk_stream = iter_chunks(row_stream, &mut builder);
+
+        if limit == 0 {
```

Comment:
Here should be `args.chunk_size`

Reply:
It is because we are enabling throttle to 0, I get that you take `min(rate_limit, args.chunk_size)` as param in chunk builder but when `rate_limit = 0`, it leads to panic.

Reply:
The `limit` here has nothing todo with rate limit, it is the parameter to the `LIMIT` in the SQL sent to upstream.